orangecanvas/application/addons.py

```diff
@@ -1276,11 +1280,16 @@ def query_pypi(names: List[str]) -> List[_QueryResult]:
 
 
 def list_available_versions(config, session=None):
-    # type: (config.Config, Optional[requests.Session]) -> List[Installable]
+    # type: (config.Config, Optional[requests.Session]) -> (List[Installable], List[Exception])
     if session is None:
         session = _session()
 
-    defaults = config.addon_defaults_list()
+    exceptions = []
+
+    try:
+        defaults = config.addon_defaults_list()
+    except requests.exceptions.RequestException as e:
+        exceptions.append(e)
```

Comment:
```suggestion
        defaults = []
        exceptions.append(e)
```
When it fails defaults remain unset and so UnboundLocalError is raised on line 1303. Set default also when package retrieval fails.